### PR TITLE
Save history as JSON, to handle multiline entries.

### PR DESCRIPTION
**Pull request by anntzer**
See #103.

### CODE REVIEW

**Comment by dsblank** [comment]
Looks good to me; thank you for this!

File: metakernel/_metakernel.py [hunk 1]
```diff
@@ -389,28 +389,17 @@ def do_history(self, hist_access_type, output, raw, session=None,
...
-            with open(self.hist_file, 'wb') as fid:
-                fid.write('')
-        with open(self.hist_file, 'rb') as fid:
-            history = fid.read().decode('utf-8', 'replace')
-        history = history.splitlines()
-        history = history[:self.max_hist_cache]
-        self.hist_cache = history
-        history = [(None, None, h) for h in history]
-        return {'history': history}
+        with open(self.hist_file) as fid:
```
Comment (blink1073) [inline metakernel/_metakernel.py]: Won't this cause an error if the file does not exist?

Comment (anntzer) [inline metakernel/_metakernel.py]: `get_history_file` (called by the constructor) makes sure that the file exists.

**Comment by blink1073** [comment]
LGTM, thanks @anntzer!

**Pull request merged**